src/floating-focus.js

```diff
@@ -164,15 +170,41 @@ export default class FloatingFocus {
 		});
 	}
 
-	repositionElement(target, floater) {
+	standardizeFloat(number) {
+		return Math.round(parseFloat(number.toFixed(3) * 1000)) / 1000;
+	}
+
+	getFloaterPosition(target) {
 		const rect = target.getBoundingClientRect();
 
+		const { width, height } = rect;
 		const left = rect.left + rect.width / 2;
 		const top = rect.top + rect.height / 2;
 
-		floater.style.left = `${left}px`;
-		floater.style.top = `${top}px`;
-		floater.style.width = `${rect.width}px`;
-		floater.style.height = `${rect.height}px`;
+		return {
+			left: `${this.standardizeFloat(left)}px`,
+			top: `${this.standardizeFloat(top)}px`,
+			width: `${this.standardizeFloat(width)}px`,
+			height: `${this.standardizeFloat(height)}px`,
+		};
+	}
+
+	monitorElementPosition() {
+		if (!this.target) {
+			return;
+		}
+
+		const newFloaterPosition = this.getFloaterPosition(this.target);
+
+		if (!isEqual( pick(this.floater.style, ['left','top','width','height']), newFloaterPosition )) {
+			this.floater.classList.add('moving');
+			Object.assign(this.floater.style, newFloaterPosition);
+			clearTimeout(this.movingTimeout);
+			this.movingTimeout = setTimeout(() => this.floater.classList.remove('moving'), MOVE_DURATION);
+		}
+	}
+
+	repositionElement(target, floater) {
+		Object.assign(floater.style, this.getFloaterPosition(target));
 	}
```

Comment:
This feels a bit like mutability magic, but that could be me or personal preference.

Maybe more something like: 
```this.floater.style = { ...floater.style, this.getFloaterPosition(target) };```

Reply:
I mean, it's also already done like this in `resolveTargetOutlineStyle`, and I think that's perfectly readable (except that here the assigned properties come from another method)... And to me it feels a bit better like this since we're not completely reassigning `this.floater.style` (including all other properties that have nothing to do with this operation)... i dunno mann....